contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -163,31 +203,23 @@ contract ThreePoolStrategy is InitializableAbstractStrategy {
      */
     function withdrawAll() external onlyVaultOrGovernor nonReentrant {
         // Withdraw all from Gauge
-        (, uint256 gaugePTokens, ) = _getTotalPTokens();
+        (, uint256 gaugePTokens, uint256 totalPTokens) = _getTotalPTokens();
         ICurveGauge(crvGaugeAddress).withdraw(gaugePTokens);
-        // Remove entire balance, 3pool strategies only support a single asset
-        // so safe to use assetsMapped[0]
-        IERC20 asset = IERC20(assetsMapped[0]);
-        uint256 pTokenBalance = IERC20(assetToPToken[address(asset)]).balanceOf(
-            address(this)
-        );
-        uint256 minWithdrawAmount = pTokenBalance.mulTruncate(
+        uint256 minWithdrawAmount = totalPTokens.mulTruncate(
             uint256(1e18).sub(maxSlippage)
         );
-        ICurvePool(platformAddress).remove_liquidity_one_coin(
-            pTokenBalance,
-            poolCoinIndex,
-            minWithdrawAmount
-        );
-        // Transfer the asset out to Vault
-        asset.safeTransfer(vaultAddress, asset.balanceOf(address(this)));
+        ICurvePool threePool = ICurvePool(platformAddress);
+        threePool.remove_liquidity(totalPTokens, minWithdrawAmount);
+        // Note that Curve will provide all 3 of the assets in 3pool even if
```

Comment:
Good thinking. Good comment explaining why we are doing something out of our ordinary.